services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -83,9 +83,15 @@ void checkFlowForEndpointConflicts(Flow requestedFlow) throws FlowValidationExce
 
 
         Optional<String> conflictedFlow = conflictsOnSource.stream()
-                .filter(flow -> !requestedFlow.getFlowId().equals(flow.getFlowId())
-                        && (flow.getSrcVlan() == 0 || flow.getSrcVlan() == requestedFlow.getSrcVlan()
+                .filter(flow -> !requestedFlow.getFlowId().equals(flow.getFlowId()))
```

Comment:
I am sorry, but this is piece of crap - extremely overcomplicated logic plus copy & pasted twice.

We have 2 point (endpoint) we need to check is there any flow that use same endpoints. So take first endpoint (flow source) and lookup this endpoint in persistent storage (it can be source or end endpoint of existing flow) if find - report conflict, if not take second endpoint and lookup it in persistent storage - report conflict if find.

So we need to translate stream of flows loaded from persistent storage into stream of endpoints, than lookup match.